pkg/apis/query/v0alpha1/query.go

```diff
@@ -32,12 +32,22 @@ func GetResponseCode(rsp *backend.QueryDataResponse) int {
 	if rsp == nil {
 		return http.StatusInternalServerError
 	}
+
```

Comment:
I would suggest looking at existing /api/ds/query logic and copy that https://github.com/grafana/grafana/blob/5ae5fa3a7a0d6d5187b38bbe8d8f9504c5bd465d/pkg/api/ds_query.go#L86

Basically 400 if one of the data responses contains an error, otherwise 200

Reply:
See https://github.com/grafana/grafana/pull/90191